### DISCUSSION TITLE
Yolov8 is good at detecting but not classifying

### DISCUSSION THREAD

**Discussion by M-Amrollahi** [ultralytics/ultralytics#5750] [Q&A]
I have trained my model based on Yolov8. The issue is that the model perform good at detecting objects and finding bboxes but the precision and recall is so low. Here is part of the results for val set:
<img width="612" alt="image" src="https://github.com/ultralytics/ultralytics/assets/44016758/566e2811-703a-4353-b6b3-2b993fa986cd">

![results](https://github.com/ultralytics/ultralytics/assets/44016758/3501753a-35e6-4eb9-92a8-d6a4e50de071)

![confusion_matrix](https://github.com/ultralytics/ultralytics/assets/44016758/58d4195b-bb96-4220-a6b6-1ce5b784aee2)

The objects are small like (20 x 20) shapes. 
What I need to do to increase the performance?

**Comment by gillonba** [comment]
I've been dealing with this myself recently.  It looks like it does not handle datasets with a small number of categories very well and likes to end training early.  How big is your validation set?  What parameters are you using?

**Reply by M-Amrollahi** [reply]
The val set is not that big. I have 5 images for validation.
[Here](https://github.com/M-Amrollahi/ultralytics/blob/main/ultralytics/cfg/default.yaml) is my fork and also my configs.

**Reply by MoAbbasid** [reply]
try to increase the number of epochs and see

**Reply by gillonba** [reply]
No good.  I've been testing classification lately and I have a dataset of about 3,000 training images and 400 validation images across two classes.  I've tried setting the patience as high as 150 and it doesn't really help.  When training, usually what I see is that I get spiky accuracy early on, in the teens, and then the accuracy flattens out and even starts to decline.  Once the patience runs out, I end up with my 'best' being from early in the training run.  I'm wondering if it would work better with more classes, since it uses both top1 and top5 to evaluate accuracy

**Reply by glenn-jocher** [reply]
@gillonba it's important to ensure that your dataset is sufficiently large and diverse, especially for the validation set. A small validation set with only 5 images may not be representative enough to accurately gauge the model's performance. Additionally, having a balanced dataset across your classes can help improve precision and recall.

If you're experiencing early stopping, it might be due to the patience parameter being too low. However, since you've already tried increasing it without success, you might want to look into other aspects such as the learning rate schedule or even the data augmentation pipeline.

Lastly, ensure that your model is not overfitting to the training data by monitoring the performance on the validation set. If the validation performance starts to decline while training performance improves, this could be a sign of overfitting.

For more detailed guidance, you can refer to our documentation on training and hyperparameter tuning at https://docs.ultralytics.com.

**Reply by MoAbbasid** [reply]
@glenn-jocher is YOLOv8 using Anchor-boxes by default? I thought it was anchorless..

**Reply by glenn-jocher** [reply]
@MoAbbasid yes that's correct, YOLOv8 has an anchor-less head, it's one of the main changes from YOLOv5.
